test/scenarios/bookshop/read.test.js

```diff
@@ -232,6 +232,31 @@ describe('Bookshop - Read', () => {
     expect(forUpdateResults).to.deep.eq([{ ID: 101 }])
   })
 
+  test('apply SELECT properties from ref', async () => {
+    const { Books } = cds.entities('sap.capire.bookshop')
+    const query = cds.ql`SELECT FROM ${Books}[
+      where ID > 210
+      group by title
+      having computed
+      order by title desc
+      limit 2
+    ] {
```

Comment:
What are the semantics of those properties if they are not part of the leaf-ref of a from clause? E.g.:

```js
cds.ql`SELECT FROM Books {
      books[
        where ID > 210
        group by title
        having computed
        order by title desc
        limit 2
    ].title as top2
  }`
```

Reply:
- `order by`, `limit`, and `group by` are only reasonable for expands
- `where` is also reasonable for inlines